### ISSUE TITLE
Enhancement: Tray and notifications

### ISSUE THREAD

**Issue by sr-tream** [FreeTubeApp/FreeTube#59]
A some ideas for this great app:
* Hide app to tray
* Notify about new videos in subscriptions
* Restore old video list after start program, and update list in background

**Comment by sr-tream** [comment]
> What benefit would this have over minimizing the window? If notifications are implemented then I can understand having it there as a notification indicator. Right now in the app's current state I do not see much of a reason to do this.

Yeah, it for notifications.
> I'm assuming you're talking about subscriptions?

Yep

**Comment by Victor239** [comment]
Found this issue, I'd also like to request the close to tray feature.

**Comment by je-vv** [comment]
I believe the tray would be pretty important, when using freetube as a music player, getting rid of it from the toolbar, but keeping it iconified in the tray, and providing control options through right click, like pause, play, stop, besides a setting option also available with right click.  That's way different than minimizing, and uses way less space, just the tray icon...

**Comment by letkan** [comment]
> I believe the tray would be pretty important, when using freetube as a music player, getting rid of it from the toolbar, but keeping it iconified in the tray, and providing control options through right click, like pause, play, stop, besides a setting option also available with right click. That's way different than minimizing, and uses way less space, just the tray icon...

I agree, tray is a useful feature to have, minimizing it just takes too much space. I wish there was an option like in nativefier when building, '--tray', or something in settings to that effect.

**Comment by saranvdev** [comment]
Agreed, takes too much space on the toolbar.

**Comment by GeniusBroccoli** [comment]
Love tray option, must have, listen many videos in the background and working, want more toolbar space

**Comment by FredericGuilbault** [comment]
+1 for this feature, 
Start a 3h playlist, minimize to systray, enjoy

**Comment by ruggeriruggero** [comment]
I love this software, I made an account specifically to +1 the minimize to icon tray request

**Comment by omega3** [comment]
> What benefit would this have over minimizing the window?

I still prefer classic task manager like [this](https://userbase.kde.org/images.userbase/1/11/Task-manager-in-panel.png
) one over icon task manager. So the issue is that when I am in web browser and do my stuff I don't want to many programs on task manager that are not used because it is a distraction of attention but I still want to have program running for actions like for example open youtube link in freetube (by redirecting link via browser add-on). Then it would maximize.

It is really legitimate request with valid use-cases.

**Comment by je-vv** [comment]
Well, that might be a matter of taste.  I actually dislike having a crowded task bar, that it's really hard to identify what you're looking for.  I prefer a tray, or now a days called notification area, nothing in the taskbar, particularly if it's something that might be run for long periods of time and you can just interact with it with pause/resume/stop/start buttons and so on.

Perhaps it's a tendency to avoid the tray, but that's somehow now days replaced but more crowded stuff...  Like on gnome and kde, now a days besides the panel, you need an extra dock, which is nothing less than another panel.  So you end up with one or two panels plus a dock.

To me the tray is the cleanest desk option.  It should work such that when you minimize it doesn't go to the task bar, as shown on your image, but instead it minimizes to tray, so not polluting the task bar.

At times it's even a matter of tendencies imposed by major DEs, which is also sad.  I'm getting feedback from other SW, that on gnome the tendency is not to have tray, and then that must be followed, as if following what gnome does would make things any better...

So it's a matter of taste, and ultimately it boils down to a developer preference.

**Comment by max-blank** [comment]
I also think this would be a cool idea. Being able to reduce/close the app in systray is useful, especially if you decide to launch it automatically when windows start!

**Comment by je-vv** [comment]
I was very interested on tray support before, though I'm no longer a proponent for that, but I didn't open this issue.  I'm now using i3 in "tabbed" with polybar, without tray, and without taskbar (just the i3 module showing me the active workspaces), and a few status, plus some buttons/icons for quick launching some apps.  I got used to no taskbar and no tray.  Just in case someone might be interested in exploring that way.  The same can be accomplished on wayland of course with sway and yambar or similar bar (yambar particularly force no tray since it doesn't support it but it's not longer maintained).
